File: guests/java/dev/kaya/milestone2kt/Radio.java
package dev.kaya.milestone2kt;

import dev.kaya.KayaApp;

/**
 * The radio conformance scene from the JVM. See
 * guests/rust/radio.rs and tools/scenes/radio.steps.
 */
final class Radio {
    private static final String[] OPTIONS = {"Small", "Medium", "Large"};

    static void app() {
        KayaApp app = new KayaApp();

        app.build(tx -> {
            tx.window(0).title("radio");
            KayaApp.Signal<String> size = tx.signal("size: Small");

            tx.mount(tx.column(() -> {
                tx.radio(OPTIONS, 0, (t, index) ->
                        t.write(size, "size: " + OPTIONS[index]));
                tx.label(size); // label#0
            }));
            return null;
        });

        app.dispatchLoop();
    }

    private Radio() {}
}

File: guests/go/radio/radio.go
// The radio conformance scene, Go port. See
// guests/rust/radio.rs and tools/scenes/radio.steps.
package radio

import (
	kaya "dev.kaya/bindings/go"
)

var options = []string{"Small", "Medium", "Large"}

func App() *kaya.App {
	app := kaya.NewApp()

	var size kaya.Signal[string]
	app.Build(func(tx *kaya.Tx) {
		tx.Window(0).Title("radio")
		size = tx.Signal("size: Small")

		tx.Mount(tx.Column(func() {
			tx.Radio(options, 0, func(tx *kaya.Tx, index int) {
				tx.Write(size, "size: "+options[index])
			})
			tx.Label(size) // label#0
		}))
	})

	return app
}
